Assets/MixedRealityToolkit.Services/InputSimulation/Implementation/InputSimulationService.cs

```diff
@@ -0,0 +1,430 @@
+                }
+            }
+        }
+
+        #endregion BaseInputDeviceManager Implementation
+
+        private MixedRealityInputSimulationProfile inputSimulationProfile = null;
+
+        /// <inheritdoc/>
+        public MixedRealityInputSimulationProfile InputSimulationProfile
+        {
+            get
+            {
+                if (inputSimulationProfile == null)
+                {
+                    inputSimulationProfile = ConfigurationProfile as MixedRealityInputSimulationProfile;
+                }
+                return inputSimulationProfile;
+            }
+            set
+            {
+                inputSimulationProfile = value;
+            }
+        }
+
+        /// <inheritdoc/>
+        IMixedRealityEyeSaccadeProvider IMixedRealityEyeGazeDataProvider.SaccadeProvider => null;
+
+        /// <inheritdoc/>
+        bool IMixedRealityEyeGazeDataProvider.SmoothEyeTracking { get; set; }
+
+        private void EnableCameraControl()
+        {
+            if (cameraControl == null)
+            {
+                cameraControl = new ManualCameraControl(InputSimulationProfile);
+            }
+        }
+
+        private void DisableCameraControl()
+        {
+            if (cameraControl != null)
+            {
+                cameraControl = null;
+            }
+        }
+
+        private void EnableHandSimulation()
+        {
+            if (handDataProvider == null)
+            {
+                handDataProvider = new SimulatedHandDataProvider(InputSimulationProfile);
+            }
+        }
+
+        private void DisableHandSimulation()
+        {
+            RemoveAllHandDevices();
+
+            if (handDataProvider != null)
+            {
+                handDataProvider = null;
+            }
+        }
+
+        private Vector3 lastMousePosition;
+        private bool wasFocused;
+        private bool wasCursorLocked;
+
+        private void ResetMouseDelta()
+        {
+            lastMousePosition = UnityEngine.Input.mousePosition;
+        }
+
+        private MouseDelta UpdateMouseDelta()
+        {
+            var profile = InputSimulationProfile;
+
+            bool isFocused = Application.isFocused;
+            bool gainedFocus = (!wasFocused && isFocused);
+            wasFocused = isFocused;
+
+            bool isCursorLocked = UnityEngine.Cursor.lockState != CursorLockMode.None;
+            bool cursorLockChanged = (wasCursorLocked != isCursorLocked);
+            wasCursorLocked = isCursorLocked;
+
+            // Reset in cases where mouse position is jumping
+            if (gainedFocus || cursorLockChanged)
+            {
+                ResetMouseDelta();
+                return new MouseDelta();
+            }
+            else
+            {
+                // Use frame-to-frame mouse delta in pixels to determine mouse rotation.
+                // The traditional GetAxis("Mouse X") method doesn't work under Remote Desktop.
```

Comment:
It feels like this comment should apply to the else statement (could we move it closer there?).

Otherwise it looks like the comment says "we shouldn't do this" and then the code immediately does it (in what seems like the typical/first thing that is read)

Reply:
Yeah, it's a confusing legacy comment. You're correct, it probably applies to the else case, moved it there.